File: program/print-pascal's-triangle/PrintPascal'sTriangle.cs
using System;

namespace PascalTriangle {
   class PascalTriangle {
      public static void Main() {
         int rows = 5, val = 1, blank, i, j;
         Console.WriteLine("Pascal's triangle");
         for(i = 0; i<rows; i++) {
            for(blank = 1; blank <= rows-i; blank++)
            Console.Write(" ");
            for(j = 0; j <= i; j++) {
               if (j == 0||i == 0)
               val = 1;
               else
               val = val*(i-j+1)/j;
               Console.Write(val + " ");
            }
            Console.WriteLine();
         }
      }
   }
}

File: program/program/print-pascal's-triangle/PrintPascal'sTriangle.rb
def print_pascals_triangle(n)
    triangle = []
  
    #Generates Pascal's triangle
    (0...n).each do |row|
        #Creates an empty array for the current row  
        triangle[row] = []  
        (0..row).each do |col|
            if col.zero? || col == row
            #Sets the boundary elements to 1  
            triangle[row][col] = 1
            else
            #Calculates the current element by summing the two elements above it
            triangle[row][col] = triangle[row - 1][col - 1] + triangle[row - 1][col]
            end
        end
    end
  
    #Prints Pascal's triangle
    triangle.each do |row|
        #Adds leading spaces to align rows based on the number of elements  
        print ' ' * (n - row.length)
        row.each do |num|
            #Prints each number followed by a space
            print "#{num} "
        end
        #Moves to the next line after printing a row
        puts  
    end
end
  
#Test
n = 5
print_pascals_triangle(n)
